### PR TITLE
Branch doc

### PR DESCRIPTION
**Pull request by adaeks**
Add branching documentation for functions in the form of comments. Part of #4.

### CODE REVIEW

File: api/src/main/java/io/opencensus/metrics/export/Metric.java [hunk 1]
```diff
@@ -103,37 +103,59 @@ private static Metric createInternal(
...
         if (value.getClass().getSuperclass() != null) { // work around nullness check
           System.out.println(Thread.currentThread().getStackTrace()[1].getMethodName() + 3);
           // AutoValue classes should always have a super class.
           valueClassName = value.getClass().getSuperclass().getSimpleName();
         }
+        
+        // enter switch block
         switch (type) {
+
+          // case without purpose
```
Comment (augustjanse) [inline api/src/main/java/io/opencensus/metrics/export/Metric.java]: Not without purpose, this is a fallthrough (keeps going until the break)

File: api/src/main/java/io/opencensus/trace/Tracestate.java [hunk 2]
```diff
@@ -229,18 +229,26 @@ public static Entry create(String key, String value) {
...
+    
+    // enter branch if the lenght of the key exceeds maximum size, return false
     if (key.length() > KEY_MAX_SIZE
         || key.isEmpty()
         || key.charAt(0) < 'a'
         || key.charAt(0) > 'z') {
       System.out.println(Thread.currentThread().getStackTrace()[1].getMethodName() + 1);
       return false;
+    
+      // else statement without purpose
```
Comment (augustjanse) [inline api/src/main/java/io/opencensus/trace/Tracestate.java]: I added this else for coverage, so you could mention that instead

File: api/src/main/java/io/opencensus/trace/Tracestate.java [hunk 3]
```diff
@@ -249,10 +257,14 @@ private static boolean validateKey(String key) {
           && c != '/') {
         System.out.println(Thread.currentThread().getStackTrace()[1].getMethodName() + 4);
         return false;
+
+      // else statement without purpose
```
Comment (augustjanse) [inline api/src/main/java/io/opencensus/trace/Tracestate.java]: For coverage

File: exporters/trace/instana/src/main/java/io/opencensus/exporter/trace/instana/InstanaExporterHandler.java [hunk 4]
```diff
@@ -125,46 +125,67 @@ private static String attributeValueToString(AttributeValue attributeValue) {
...
+        // enter for-loop if the attributeMap contains a entrySet
         for (Entry<String, AttributeValue> entry : attributeMap.entrySet()) {
           System.out.println(Thread.currentThread().getStackTrace()[1].getMethodName() + 7);
+          
+          // enter branch if the lenght of the stringbuilder data.Sb i greater than one.
           if (dataSb.length() > 1) {
             System.out.println(Thread.currentThread().getStackTrace()[1].getMethodName() + 8);
             dataSb.append(',');
+
+            // else-statement without a purpose
```
Comment (augustjanse) [inline exporters/trace/instana/src/main/java/io/opencensus/exporter/trace/instana/InstanaExporterHandler.java]: Coverage

**Review by augustjanse — CHANGES_REQUESTED** [review]
Looks good, please fix comments!

File: api/src/main/java/io/opencensus/metrics/export/Metric.java (hunk 1 above)
Comment (augustjanse) [reply api/src/main/java/io/opencensus/metrics/export/Metric.java]: Also later

Comment (adaeks) [reply api/src/main/java/io/opencensus/metrics/export/Metric.java]: Hmm... but how does it have a purpose? If it enters the case, no statements gets executed, and same thing happens if the case never gets entered.

**Review by augustjanse — APPROVED** [review]
Thanks!

**Pull request merged**